### PR TITLE
[Issue #1510]: Split FE / BE docker-compose

### LINKED ISSUE #1510: [Task]: Split top-level docker-compose file into a front-end and back-end file to make running each individually simpler
### Summary

https://github.com/HHS/simpler-grants-gov/blob/main/docker-compose.yml starts up:
- The API
- The DB for the API
- The Next JS app
- Storybook

But if I'm an API developer, I likely don't need those last two most of the time, and vice-versa for the front-end developers. While running them altogether should be possible / easy, we should also make running them separately easy. We can do this by using https://docs.docker.com/compose/multiple-compose-files/include/

We make two docker-compose files, one in the frontend, and one in the API folder. Then we make a top-level one that includes both of those use include.

### Acceptance criteria

- [ ] Docker-compose files created and able to start up the API/frontend without the other uneventfully

### PR DESCRIPTION
**Pull request by rylew1**
## Summary
Fixes #1510

## Changes proposed
- Move `nextjs` and `storybook` service to `./frontend/docker-compose.yml`
- Move `grants-api` and `grants-db` service to `./api/docker-compose.yml`
- Top level `./docker-compose.yml`

## Demo:

- Command from FE folder => `make build-dev dev storybook`
- Command from BE folder => `make build start run-logs`

image

### CODE REVIEW

**Review by coilysiren — APPROVED** [review]
👍!

File: docker-compose.yml [hunk 1]
```diff
@@ -1,65 +1,8 @@
...
-      - ./frontend/.storybook:/frontend/.storybook
-      - ./frontend/stories:/frontend/stories
-    restart: always
-    ports:
-      - 6006:6006
-
-volumes:
-  grantsdbdata:
+# services:
+  # Define other services or override configurations
```
Comment (rylew1) [inline docker-compose.yml:8]: could remove this

File: docker-compose.yml [hunk 2]
```diff
@@ -1,65 +1,8 @@
 version: '3'
```
Comment (rylew1) [inline docker-compose.yml:1]: how do we check this updated file runs properly - via a github action?

**Comment by rylew1** [comment]
When you start all containers the frontend gives an error trying to connect to the api `(Error: connect ECONNREFUSED 127.0.0.1:8080),`

But this is also happening on main

**Review by chouinar — APPROVED** [review]

**Pull request merged**